set.go

```diff
@@ -55,9 +54,9 @@ func (s *Set) Visit(c *Cid) bool {
 
 // ForEach allows to run a custom function on each
 // Cid in the set.
-func (s *Set) ForEach(f func(c *Cid) error) error {
+func (s *Set) ForEach(f func(c Cid) error) error {
 	for cs := range s.set {
-		c, _ := Cast([]byte(cs))
+		c, _ := Cast(cs.Bytes())
```

Comment:
We can just use the cid as-is (this is a left-over from when we needed to reconstruct it).